x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -580,7 +540,92 @@ export const StepDetailsForm: FC<Props> = React.memo(
                   data-test-subj="transformContinuousDelayInput"
                 />
               </EuiFormRow>
-            </Fragment>
+            </>
+          )}
+
+          {/* Retention policy */}
+          <EuiFormRow
+            helpText={
+              isRetentionPolicyAvailable === false
+                ? i18n.translate('xpack.transform.stepDetailsForm.retentionPolicyError', {
+                    defaultMessage:
+                      'Retention policy settings are not available for indices without date fields.',
+                  })
```

Comment:
Not sure if it should be mentioned in `helpText`.  Would the info callout be a better place?

Reply:
The messaging here LGTM - the field is present, but disabled, with the help text explaining the reason.

![image](https://user-images.githubusercontent.com/7405507/107937865-72591280-6f7c-11eb-90cf-8ad613a5c336.png)